packages/core/src/lib/markdown-it/highlight/HighlightRuleComponent.ts

```diff
@@ -62,9 +65,13 @@ export class HighlightRuleComponent {
     }
 
     // Match line-number syntax
-    if (!Number.isNaN(compString)) { // ensure the whole string can be converted to number
-      const lineNumber = parseInt(compString, 10) + lineNumberOffset;
```

Comment:
parseInt doesn't handle cases such as "123abc" well it will simply extract the 123 out, I believe the intention is to consider it invalid.

Reply:
Could we review the number checking methods in this file and consolidate it into a single method to be used in the few places inside the file? I observed that the checks are not consistently done nor written in the exact same way, when they could be. What do you think?

Reply:
@zlimez I think you might have missed this comment. To be clear, there's a few places in this method that does the check to ensure that a converted number is a valid line number. How about we extract that logic into a method to be reused if needed, to ensure that the line number is valid?

Reply:
Yap sorry I missed the comment yap will do